File: app/src/main/java/com/hidesign/ported/ui/home/PastTripFragment.java
package com.hidesign.ported.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;
import com.hidesign.ported.Functions;
import com.hidesign.ported.R;
import com.hidesign.ported.adapters.RecyclerAdapter;
import com.hidesign.ported.models.Trips;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import timber.log.Timber;

public class PastTripFragment extends Fragment {

    private static final String TAG = "HomeFragment";
    private RecyclerView mRecyclerView;
    private List<Trips> pastTrips = new ArrayList<>();
    private Functions func = new Functions();
    private DatabaseReference mDatabase;
    private View mProgressBar;

    public PastTripFragment(){}

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View viewInflate = inflater.inflate(R.layout.fragment_past_trips, container, false);

        Objects.requireNonNull(((AppCompatActivity) requireActivity()).getSupportActionBar()).show();

        mProgressBar = viewInflate.findViewById(R.id.progress_bar);

        mRecyclerView = viewInflate.findViewById(R.id.recyclerView);
        mRecyclerView.setHasFixedSize(true);
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getActivity());
        mRecyclerView.setLayoutManager(mLayoutManager);

        FirebaseAuth mAuth = FirebaseAuth.getInstance();
        FirebaseUser currentUser = mAuth.getCurrentUser();
        if (currentUser != null) {
            mDatabase = FirebaseDatabase.getInstance().getReference().child(currentUser.getUid());
        }

        return viewInflate;
    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        mDatabase.child("Trips").addListenerForSingleValueEvent(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                for (DataSnapshot postSnapshot: dataSnapshot.getChildren()) {
                    Trips post = postSnapshot.getValue(Trips.class);
                    pastTrips.add(post);
                }
                func.sortList(pastTrips);
                showList(pastTrips);
                mProgressBar.setVisibility(View.GONE);
            }
            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {

            }
        });
    }

    private void showList(List<Trips> display) {
        Timber.tag(TAG).e("showList: %s", pastTrips.size());

        RecyclerAdapter mAdapter = new RecyclerAdapter(display.size(), R.layout.item_trip);
        mAdapter.setOnRecyclerAdapterListener((adapter, v, position) -> {

            TextView _Date = v.view.findViewById(R.id.Date);
            _Date.setText(func.getDate(display.get(position).getTripDate(), "dd MMM yyyy HH:mm"));

            TextView _StartAddress = v.view.findViewById(R.id.StartLocation);
            _StartAddress.setText((display.get(position).getStartAddress()));

            TextView _EndAddress = v.view.findViewById(R.id.EndLocation);
            _EndAddress.setText(display.get(position).getEndAddress());

            List<String> distance = func.calculateDistance(display.get(position).getTripDistance());
            TextView _Distance = v.view.findViewById(R.id.Distance);
            _Distance.setText(distance.get(0));
            TextView _DistanceSystem = v.view.findViewById(R.id.DistanceSystem);
            _DistanceSystem.setText(distance.get(1));

            v.view.setOnClickListener(v1 -> {

            });
        });
        mRecyclerView.setAdapter(mAdapter);
    }
}

File: app/src/main/java/com/hidesign/ported/ui/home/PastTripFragment.kt
package com.hidesign.ported.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*
import com.hidesign.ported.Functions
import com.hidesign.ported.R
import com.hidesign.ported.adapters.RecyclerAdapter
import com.hidesign.ported.models.Trips
import timber.log.Timber
import java.util.*

class PastTripFragment : Fragment() {

    private lateinit var mRecyclerView: RecyclerView
    private val pastTrips: MutableList<Trips> = ArrayList()
    private val func = Functions()
    private var mDatabase: DatabaseReference? = null
    private var mProgressBar: View? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val viewInflate = inflater.inflate(R.layout.fragment_past_trips, container, false)

        (requireActivity() as AppCompatActivity).supportActionBar?.show()

        mProgressBar = viewInflate.findViewById(R.id.progress_bar)

        mRecyclerView = viewInflate.findViewById(R.id.recyclerView)
        mRecyclerView.setHasFixedSize(true)
        val mLayoutManager: RecyclerView.LayoutManager = LinearLayoutManager(activity)
        mRecyclerView.layoutManager = mLayoutManager

        val currentUser = FirebaseAuth.getInstance().currentUser
        if (currentUser != null) {
            mDatabase = FirebaseDatabase.getInstance().reference.child(currentUser.uid)
        }

        return viewInflate
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mDatabase!!.child("Trips").addListenerForSingleValueEvent(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                for (postSnapshot in dataSnapshot.children) {
                    val post = postSnapshot.getValue(Trips::class.java)
                    pastTrips.add(post!!)
                }
                func.sortList(pastTrips)
                showList(pastTrips)
                mProgressBar!!.visibility = View.GONE
            }

            override fun onCancelled(databaseError: DatabaseError) {}
        })
    }

    private fun showList(display: List<Trips?>) {
        Timber.e("showList: %s", pastTrips.size)
        val mAdapter = RecyclerAdapter(display.size, R.layout.item_trip)
        mAdapter.setOnRecyclerAdapterListener { _: RecyclerAdapter?, v: RecyclerAdapter.ViewHolder, position: Int ->
            val date = v.view.findViewById<TextView>(R.id.Date)
            date.text = func.getDate(display[position]!!.tripDate, "dd MMM yyyy HH:mm")

            val startAddress = v.view.findViewById<TextView>(R.id.StartLocation)
            startAddress.text = display[position]!!.startAddress
            val endAddress = v.view.findViewById<TextView>(R.id.EndLocation)
            endAddress.text = display[position]!!.endAddress

            val distance = func.calculateDistance(display[position]!!.tripDistance)
            val distanceDisplay = v.view.findViewById<TextView>(R.id.Distance)
            distanceDisplay.text = distance[0]
            val distanceSystem = v.view.findViewById<TextView>(R.id.DistanceSystem)
            distanceSystem.text = distance[1]
        }
        mRecyclerView.adapter = mAdapter
    }
}